src/lightning_app/core/app.py

```diff
@@ -506,7 +503,7 @@ def _run(self) -> bool:
 
             self._update_run_time_monitor()
 
-            if self.ready and self._has_updated and self.should_publish_changes_to_api and self.api_publish_state_queue:
+            if self._has_updated and self.should_publish_changes_to_api and self.api_publish_state_queue:
                 self.api_publish_state_queue.put(self.state_vars)
```

Comment:
Alternatively, let's pass a tuple there (state, status) even if status can be inferred from state.

Reply:
Went for the tuple approach to avoid adding too much into state